tests/structured_conf/test_structured_config.py

```diff
@@ -895,24 +905,18 @@ def test_str2str_with_field(self, class_type: str) -> None:
         with pytest.raises(KeyValidationError):
             cfg[Color.RED] = "fail"
 
-    def test_str2int_with_field_of_different_type(self, class_type: str) -> None:
-        module: Any = import_module(class_type)
-        cfg = OmegaConf.structured(module.DictSubclass.Str2IntWithStrField())
-        assert cfg.foo == "bar"
-
-        cfg.one = 1
-        assert cfg.one == 1
-
-        with pytest.raises(ValidationError):
-            # bad
-            cfg.hello = "world"
-
     class TestErrors:
         def test_usr2str(self, class_type: str) -> None:
             module: Any = import_module(class_type)
             with pytest.raises(KeyValidationError):
                 OmegaConf.structured(module.DictSubclass.Error.User2Str())
 
+        def test_str2int_with_field_of_different_type(self, class_type: str) -> None:
+            module: Any = import_module(class_type)
+            cfg = OmegaConf.structured(module.DictSubclass.Error.Str2IntWithStrField())
+            with pytest.raises(ValidationError):
+                cfg.foo = "str"
```

Comment:
This tests the assignment of an incorrect type to an existing field (once you add `foo: int = 10` to the class.
```
cfg.foo = "str"
```

This tests creation of a new field with an incorrect type:
```
cfg.foo2 = "str"
```